segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -662,7 +689,14 @@ public boolean supportsAtomicWrites() {
                                 LoggerHelpers.traceLeave(log, "exists", traceId, retValue);
                                 return retValue;
                             }, executor),
-                    executor);
+                    executor)
+                    .handleAsync((v, e) -> {
+                        if (null != e) {
+                            log.error("{} exists - exception segment={}.", logPrefix, streamSegmentName, e);
```

Comment:
Can we add some args to `handleException` (like "String operationName", "String segmentName", "Object... other") and make the logging in `handleException`? That will save you from having to add so many `log.error` calls everywhere.

Reply:
Also please remove `exception` from this messge. You are already logging at `ERROR` level and then logging the exception itself afterwards.

Reply:
And I'm going to argue this should be a `WARN` with just the exception name (no stack trace). Whomever is waiting for this future upstream will re-log the exception. You'll end up logging the same stack trace twice which is very annoying when looking at logs. 

I recommend logging this as `WARN` and then call `ex.toString()` to only output the exception type and message - but no stack trace.